packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2747,14 +2749,23 @@ export class ContainerRuntime
 	/**
 	 * Process an empty batch, which will execute expected actions while processing even if there are no messages.
 	 * This is a separate function because the processCore function expects at least one message to process.
+	 * It is expected to happen only when the outbox produces an empty batch due to a resubmit flow.
```

Comment:
Maybe add `@remarks` that we don't emit the "op" event (even though we currently do for other ops that don't result in any changes to CR, like "system ops").  This is a unique case and I think it's the right call to skip it.  And we may change that policy on the other ops too.